violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/AddNodeBehavior.java

```diff
@@ -62,26 +89,65 @@ public void onMouseClicked(MouseEvent event)
         INode prototype = (INode) selectedTool.getNodeOrEdge();
         INode newNode = (INode) prototype.clone();
         newNode.setId(new Id());
-        
+
         boolean added = addNodeAtPoint(newNode, newNodeLocation);
+
         if (added)
         {
             selectionHandler.setSelectedElement(newNode);
-            
-            if (!KeyModifierUtil.isCtrl(event)) {
-	            selectionHandler.setSelectedTool(GraphTool.SELECTION_TOOL);
-	            graphToolsBar.setSelectedTool(GraphTool.SELECTION_TOOL);
-	            graphToolsBar.getAWTComponent().invalidate();
+
+            if (!KeyModifierUtil.isCtrl(event))
+            {
+                selectionHandler.setSelectedTool(GraphTool.SELECTION_TOOL);
+                graphToolsBar.setSelectedTool(GraphTool.SELECTION_TOOL);
+                graphToolsBar.getAWTComponent().invalidate();
             }
-            
+
             editorPart.getSwingComponent().invalidate();
+            if (checkIfNumberOfInitialNodesIsMoreThanOne(newNode))
+            {
+                showWarningAboutNumberOfInitialNodes();
+            }
+        }
+    }
+
+    private boolean checkIfNumberOfInitialNodesIsMoreThanOne(INode newNode)
+    {
+        if (newNode instanceof IInitialStateNode)
+        {
+            int numberOfInitialNodes = getNumberOfInitialNodes();
+            if (numberOfInitialNodes > maxNumberOfInitialNodes)
+            {
+                return true;
```

Comment:
I don't like double returns in one method.
It should be better, if you change it to one return.
boolean result = false;
if(...) {
result = true;
}
return result;